valhalla/sif/dynamiccost.h

```diff
@@ -1035,6 +1036,12 @@ class DynamicCost {
 
     // shortest ignores any penalties in favor of path length
     c.cost *= !shortest_;
+
+    // switching into/out-of an hov-lane is just a lane change.
+    bool hov_transition = (pred->is_hov_only() && !edge->is_hov_only()) ||
+                          (!pred->is_hov_only() && edge->is_hov_only());
+    c.cost *= !hov_transition;
```

Comment:
i worry this is too broad. it basically says if we are getting onto or leaving an HOV edge then its always free no matter what other factors are at play. imagine you cross a country border where the road becomes or stops being HOV. then the penalty for that border crossing is removed. this would not be expected.

for the example you cited, i think the right change would be to check if the costing has HOT enabled, and if so dont do the toll penalty when its a transition onto an HOV only toll edge

Reply:
@ktatterso yeah zero is a bit much.  Did you try to reduce it versus making it zero?

Reply:
@ktatterso maybe we can check for HOT and if we are transitioning onto a ramp we can reduce the cost.